yt_dlp/extractor/ninenow.py

```diff
@@ -59,26 +55,36 @@ def _real_extract(self, url):
             cache = page_data.get(kind, {}).get('%sCache' % kind, {})
             if not cache:
                 continue
-            common_data = (cache.get(current_key) or list(cache.values())[0])[kind]
+            common_data = {
+                'episode': (cache.get(current_key) or list(cache.values())[0])[kind],
+                'season': (cache.get(current_key) or list(cache.values())[0]).get('season', None)
+            }
             break
         else:
             raise ExtractorError('Unable to find video data')
 
-        video_data = common_data['video']
-
-        brightcove_id = video_data.get('brightcoveId') or 'ref:' + video_data['referenceId']
-        video_id = compat_str(video_data.get('id') or brightcove_id)
-
-        if not self.get_param('allow_unplayable_formats') and video_data.get('drm'):
-            self.report_drm(video_id)
-
-        title = common_data['name']
+        # Video Data extraction
+        if try_get(common_data, lambda x: x['episode']['video']['drm'], bool):
+            raise ExtractorError('This video is DRM protected.', expected=True)
+        brightcove_id = \
+            try_get(
+                common_data, 
+                lambda x: x['episode']['video']['brightcoveId'] or f"ref:{x['episode']['video']['referenceId']}"
+                , compat_str
+            ) 
+        video_id = try_get(common_data, lambda x: x['episode']['video']['id'] or brightcove_id, compat_str)
 
+        # Episode/Season data extraction
+        title = try_get(common_data, lambda x: x['episode']['name'], compat_str)
+        season_number = try_get(common_data, lambda x: x['season']['seasonNumber'], int)
+        episode_number = try_get(common_data, lambda x: x['episode']['episodeNumber'], int)
+        timestamp = unified_timestamp(try_get(common_data, lambda x: x['episode']['airDate'], compat_str) or None)
+        upload_date = unified_strdate(try_get(common_data, lambda x: x['episode']['availability'], compat_str) or None)
```

Comment:
Do these point to the same time? if so, remove `upload_date`. If not, see if `release_timestamp`/`release_date` is more appropriate for one of them

Reply:
There isn't a `release_timestamp` listed in the output template fields but I've changed `upload_date` to `release_date` which better suits.